### PR TITLE
Lint fixes, adding docs for all public functions, utilize constants and make looping consistent.

### PR DESCRIPTION
**Pull request by deckarep**
What a fantastic cache solution you've built.  Really liking the design of this cache solution!

I hope this change is welcomed...as it mainly cleans up some minor things that my linter found, makes docs consistent and adds constants.

Please, let me know if anything should be tweaked to your liking.

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -9,18 +9,24 @@ import (
 )
 
 const (
-	minBufSize = 512 * 1024
+	// segmentSize represents the number of segments within a freecache instance.
+	segmentSize = 256
```
Comment (coocood) [inline cache.go]: I think `segmentCount` or `numSegments` is better.

Comment (deckarep) [reply cache.go]: Sure, I renamed it to `segmentCount` to be consistent.

**Comment by coocood** [comment]
Nice cleanup, thank you!

**Comment by deckarep** [comment]
@coocood - thank you as well for this library and for the quick response!

**Pull request merged**